src/tables/maxp.rs

```diff
@@ -0,0 +1,47 @@
+//! The 'maxp' table.
+
+use read_fonts::tables::maxp::Maxp;
+use xml_builder::XMLElement;
+
+use crate::ttx::Ttx;
+
+impl Ttx for Maxp<'_> {
+    fn write_ttx(&self, mut into: &mut dyn std::io::Write) -> Result<(), crate::error::Error> {
+        let mut root = XMLElement::new("maxp");
+        add_u16_child(&mut root, "numGlyphs", self.num_glyphs());
```

Comment:
I think you forgot to also emit the maxp's tableVersion

Reply:
also, maxp version 0.5 (for CFF-flavored OpenType fonts) only has version and numGlyphs, and that's it; only maxp 1.0 has all the TrueType-specific fields
https://learn.microsoft.com/en-us/typography/opentype/spec/maxp

Reply:
oh, I see, you're already implicitly omitting the TT-only fields when version is 0.5, because `add_u16_child` would return `None` early in that case, thanks to the `?` in `let value: u16 = value.into()?;` (line 42)...

Reply:
https://github.com/cmyr/ttx-rs/pull/4

Reply:
maybe we could generalize this pattern of "add a simple child to root with given name and value or do nothing if it's None", and replace the concrete `add_u16_child` which only works on `u16` with a more generic method like

```rust
pub fn maybe_add_child<T: std::fmt::Display>(
    root: &mut XMLElement,
    name: &str,
    value: Option<T>,
) -> Option<T> {
    let value: T = value?;
    root.add_child(value_elem(name, &value)).unwrap();
    Some(value)
}
```

... then inside maxp's `write_ttx` we use this when read-fonts table getters return an optional value

```rust
maybe_add_child(&mut root, "maxPoints", self.max_points());
```